worker/raft/rafttransport/worker_test.go

```diff
@@ -142,9 +143,10 @@ func (s *WorkerValidationSuite) testValidateError(c *gc.C, f func(*rafttransport
 
 type WorkerSuite struct {
 	workerFixture
-	server *httptest.Server
-	worker *rafttransport.Worker
-	reqs   chan apiserver.DetailsRequest
+	server  *httptest.Server
+	worker  *rafttransport.Worker
+	reqsMtx sync.Mutex
```

Comment:
you shouldn't need a mutex around a channel, because it is a synchronization point.
You'd only need it if we are changing the object dynamically.
And we should always be having SetUpTest start after everything has torn down from the previous test.